sdk/src/wallet/operations/transaction/build_transaction.rs

```diff
@@ -48,6 +52,34 @@ where
             }
 
             inputs.push(Input::Utxo(UtxoInput::from(*input.output_id())));
+
+            if let Some(reward) = selected_transaction_data.mana_rewards.get(input.output_id()) {
+                mana_rewards += *reward;
+                context_inputs.insert(ContextInput::from(RewardContextInput::new(idx as _)?));
+                if !context_inputs.iter().any(|c| c.kind() == CommitmentContextInput::KIND) {
```

Comment:
Can you fit that into L110?

Reply:
Avoid loops and better to have it in a single place

Reply:
I'd rather have a variable like in the other PR

Reply:
fixed it